scheduler/filter/filter.go

```diff
@@ -55,15 +55,28 @@ func New(names []string) ([]Filter, error) {
 
 // ApplyFilters applies a set of filters in batch.
 func ApplyFilters(filters []Filter, config *cluster.ContainerConfig, nodes []*node.Node) ([]*node.Node, error) {
-	var err error
+	candidates, err := applyFilters(filters, config, nodes, true)
```

Comment:
that's not at all what we did before.
currently we try with hard and soft and if it doesn't work, we drop to soft (and try only with the hard)